src/ui/dialogs/spring_pane.rb

```diff
@@ -392,11 +322,21 @@ def update_node_position(node, position)
   # compilation / simulation logic:
 
   def simulate
-    compile
-    simulation_time = Benchmark.realtime do
-      @simulation_data = SimulationRunnerClient.get_hub_time_series(@force_vectors)
+    Sketchup.active_model.start_operation('compile simulation', true)
+    SimulationRunnerClient.update_model(
+        JsonExport.graph_to_json(nil, [], constants_for_springs)
+    ) do |timeseries_data, user_stats|
+      @simulation_data = timeseries_data
+      mounted_users.keys.each do |node_id|
+        stats = user_stats[node_id.to_s]
+        stats = DEFAULT_STATS if stats == {}
```

Comment:
I think we should initialize the `DEFAULT_STATS` at the beginning of the method, if we only use them there.
Further comment: I don't get what they are for, to me it feels like they are hiding that if the server does not respond with the correct user stats (maybe because the request was incorrect), we replace empty stats with invalid stats, which then probably breaks in the UI. Should we throw an exception right here?